include/picongpu/plugins/PluginController.hpp

```diff
@@ -82,119 +59,26 @@
-                    = particles::traits::SpeciesEligibleForSolver<Species, Solver>::type::value;
-            };
-        };
-
-        /* define stand alone plugins */
-        using StandAlonePlugins = pmacc::mp_list<
-            Checkpoint,
-            EnergyFields,
-            ChargeConservation,
-            SumCurrents
-#if(SIMDIM == DIM3 && PIC_ENABLE_FFTW3 == 1 && ENABLE_OPENPMD == 1)
-            ,
-            plugins::multi::Master<plugins::shadowgraphy::Shadowgraphy>
-#endif
-#if(ENABLE_OPENPMD == 1)
-            ,
-            plugins::binning::BinningDispatcher,
-            plugins::multi::Master<openPMD::openPMDWriter>
-#endif
-#if(ENABLE_ISAAC == 1) && (SIMDIM == DIM3)
-            ,
-            isaacP::IsaacPlugin
-#endif
-            >;
-
-        using AllFields = pmacc::mp_list<FieldB, FieldE, FieldJ>;
-
-        /* define species plugins */
-        using UnspecializedSpeciesPlugins = pmacc::mp_list<
-            plugins::multi::Master<EnergyParticles<boost::mpl::_1>>,
-            plugins::multi::Master<CalcEmittance<boost::mpl::_1>>,
-            plugins::multi::Master<BinEnergyParticles<boost::mpl::_1>>,
-            CountParticles<boost::mpl::_1>,
-            PngPlugin<Visualisation<boost::mpl::_1, PngCreator>>,
-            plugins::transitionRadiation::TransitionRadiation<boost::mpl::_1>
-
-#if(ENABLE_OPENPMD == 1)
-            ,
-            plugins::radiation::Radiation<boost::mpl::_1>,
-            plugins::multi::Master<ParticleCalorimeter<boost::mpl::_1>>,
-            plugins::multi::Master<PhaseSpace<particles::shapes::Counter::ChargeAssignment, boost::mpl::_1>>,
-            PerSuperCell<boost::mpl::_1>
-#endif
-            >;
-
-        using CombinedUnspecializedSpeciesPlugins
-            = pmacc::AllCombinations<VectorAllSpecies, UnspecializedSpeciesPlugins>;
-
-        using CombinedUnspecializedSpeciesPluginsEligible
-            = pmacc::mp_copy_if<CombinedUnspecializedSpeciesPlugins, TupleSpeciesPlugin::IsEligible>;
-
-        using SpeciesPlugins
-            = pmacc::mp_transform<TupleSpeciesPlugin::Apply, CombinedUnspecializedSpeciesPluginsEligible>;
-
-        /* create sequence with all fully specialized plugins */
-        using AllPlugins = MakeSeq_t<StandAlonePlugins, SpeciesPlugins>;
+        std::vector<std::shared_ptr<ISimulationPlugin>> plugins;
 
         /**
          * Initializes the controller by adding all user plugins to its internal list.
          */
         virtual void init()
         {
-            meta::ForEach<AllPlugins, PushBack<boost::mpl::_1>> pushBack;
-            pushBack(plugins);
+            // get all plugins from plugin registry
+            auto const pluginFactories = PluginRegistry::get().getPluginFactories();
+            plugins.reserve(pluginFactories.size());
+            for(auto const& pluginFactory : pluginFactories)
+            {
+                plugins.emplace_back(std::move(pluginFactory->createPlugin()));
+            }
```

Comment:
Why are you insisting on `.resize` upfront and move semantics within? We're handling a very likely tiny amount of shared pointers (tiny compared to anything where `resize` would make a difference). A significant simplification would be:
```suggestion
            // get all plugins from plugin registry
            auto const pluginFactories = PluginRegistry::get().getPluginFactories();
            std::transform(pluginFactories.cbegin(), pluginFactories.cend(), plugins.back_insert_iterator(), [](auto const& factory){return factory->createPlugin();});
```

Reply:
`reserve()` is called to avoid resizing the vector with each insertion. 

I will use your transform sugestion

Reply:
I kept the for loop. Getting the back inserter is a very ugly syntax, over all the code becomes very ugly compared to a for loop.

Reply:
Okay, I agree that that's not the most beautiful code. But we could do a `.resize()` upfront and then use `plugins.begin()`. (This time I really mean `.resize()` and not `.reserve()`.) My point was that even if we have 100 plugins, the reallocations are not an issue because it's not in a hot loop.

Reply:
`resize` would add invalid objects into the container. `reserve()` is allocating enouph space for the coming plugins and always guarantee a clean valid state of the system.